code/game/objects/items/storage/backpack.dm

```diff
@@ -350,6 +350,65 @@
 	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
 	STR.max_combined_w_class = 30
 
+/obj/item/storage/backpack/duffelbag/cursed
+	name = "living duffelbag"
+	desc = "A cursed clown duffelbag which hungers, maybe putting some food in it will stop it from eating you! Or maybe you can poison it this way..."
+	icon_state = "duffel-curse"
+	inhand_icon_state = "duffel-curse"
+	slowdown = 1.3
+	///counts time passed since it ate food
+	var/hunger = 0
+	///life left before it dies
+	var/hp = 100
+
+/obj/item/storage/backpack/duffelbag/cursed/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj,src)
+	ADD_TRAIT(src, TRAIT_NODROP, "duffelbag")
+
+/obj/item/storage/backpack/duffelbag/cursed/process()
+	///don't process if it's somehow on the floor
+	if(!iscarbon(src.loc))
+		return
+	var/mob/living/carbon/user = src.loc
+	///check hp
+	if(hp < 0)
+		user.dropItemToGround(src, TRUE)
+		var/datum/component/storage/ST = GetComponent(/datum/component/storage)
+		ST.do_quick_empty()
+		var/turf/T = get_turf(user)
+		playsound(T, 'sound/effects/splat.ogg', 50, TRUE)
+		new /obj/effect/decal/cleanable/vomit(T)
+		qdel(src)
+	hunger++
+	///check hunger
+	if((hunger > 50) && prob(20))
+		for(var/I in contents)
+			if(istype(I, /obj/item/food))
```

Comment:
You'll want to use IS_EDIBLE for this so we support both type and component food as things transition

Reply:
damn it orange man